File: experimental data/RQ2/translation_result/fail/c/cpw/gpt4/Hinted/few-shot-False/cot-ast/candidates-1/program_octal_decimal_conversion/program_octal_decimal_conversion.rs

fn program_octal_decimal_conversion(n: i32) -> i32 {
    let mut num = n;
    let mut dec_value: i32 = 0;
    let mut base: i32 = 1;

    while num != 0 {
        let last_digit = num % 10;
        num /= 10
        dec_value = dec_value.wrapping_add(last_digit * base);
        base = base.wrapping_mul(8);
    }

    dec_value
}

File: benchmark/c/cpw/program_octal_decimal_conversion/program_octal_decimal_conversion.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int program_octal_decimal_conversion ( int n ) ;
int program_octal_decimal_conversion ( int n ) {
  int num = n;
  int dec_value = 0;
  int base = 1;
  int temp = num;
  while ( temp ) {
    int last_digit = temp % 10;
    temp = temp / 10;
    dec_value += last_digit * base;
    base = base * 8;
  }
  return dec_value;
}


